src/ui/public/chrome/directives/kbn_chrome.js

```diff
@@ -43,6 +43,29 @@ export default function (chrome, internals) {
         // and some local values
         chrome.httpActive = $http.pendingRequests;
         $scope.notifList = require('ui/notify')._notifs;
+
+        // App switcher functionality.
+        function updateAppSwitcher(isOpen) {
+          $scope.appSwitcherButton = {
+            classes: isOpen ? 'app-switcher-link--close' : undefined,
+            title: isOpen ? 'Collapse' : 'Expand',
+            tooltip: isOpen ? 'Collapse side bar' : 'Expand side bar',
+          };
+        }
+
+        $scope.isAppSwitcherOpen = appSwitcherState.isOpen();
+        updateAppSwitcher($scope.isAppSwitcherOpen);
+
+        $scope.toggleAppSwitcher = event => {
+          event.preventDefault();
+          $scope.isAppSwitcherOpen = !$scope.isAppSwitcherOpen;
+          appSwitcherState.setOpen($scope.isAppSwitcherOpen);
+          updateAppSwitcher($scope.isAppSwitcherOpen);
+
+          // Notify visualizations, e.g. the dashboard, that they should re-render.
+          $scope.$root.$broadcast('ready:vis');
```

Comment:
In the `grid`, used for dashboards, you can see that both the window `resize` event and this event are handled by the same layout function: https://github.com/elastic/kibana/blob/master/src/core_plugins/kibana/public/dashboard/directives/grid.js#L102

I tested it out and it has the desired result of re-rendering a dashboard. It doesn't look like it breaks any of the other plugins which listen to it (Discover and Visualize). What do you think? I feel pretty good about using it.